tools/xtask/src/build.rs

```diff
@@ -147,6 +147,23 @@ fn build_kernel<'a>(
     crate::print_status_line("collection: kernel", Some(build_config));
     let packages = locate_packages(workspace, Some("kernel"));
     let mut options = CompileOptions::new(workspace.config(), mode)?;
+
+    if !build_config.is_default_target() {
+        let triple = Triple::new(
+            build_config.arch,
+            build_config.machine,
+            crate::triple::Host::None,
+        );
+
+        let mut target_spec = triple.to_string();
+        target_spec.insert_str(0, "src/kernel/target-spec/");
+        target_spec.push_str(".json");
+
+        let bc = BuildConfig::new(workspace.config(), None, false, &[target_spec], mode)?;
+
+        options.build_config = bc;
+    }
```

Comment:
Why is this not necessary for x86 (default)? It's been a minute since I looked at this.

Reply:
I believe when the workspace is initialized cargo detects that x86 is the default target. That is specified here: https://github.com/twizzler-operating-system/twizzler/blob/89b8dfbafbd4838449bca34decd0612c631995c8/src/kernel/.cargo/config.toml#L18-L19

The issue was getting cargo to pick the right thing for ARM. There might be a better way, but this is what I came up with after digging through some of the APIs.

Reply:
Does [build] support doing [build.x86-triple]? Maybe we could add another one of those?

Reply:
I don't think it does according to https://doc.rust-lang.org/cargo/reference/config.html#buildtarget . I just tried adding multiple triples to target and it seemed to compile all of them.

Reply:
But I'm pretty sure we just want to compile for a single target. We can also control it through the `CARGO_BUILD_TARGET` environment variable.